clients/feeder/feeder.go

```diff
@@ -211,9 +203,10 @@ func (c *Client) StateUpdate(ctx context.Context, blockNumber uint64) (*StateUpd
 	if err != nil {
 		return nil, err
 	}
+	defer body.Close()
```

Comment:
We dont defer statements that can return an error. Can you please explicitly close the body and handle the error on each branch that `return`?